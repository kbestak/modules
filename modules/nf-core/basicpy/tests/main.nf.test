nextflow_process {

    name "Test Process BASICPY"
    script "../main.nf"
    process "BASICPY"

    tag "modules"
    tag "modules_nfcore"
    tag "basicpy"

    test("basicpy - ome-tif") {

        when {
            process {
                """
                input[0] = [
                                [ id:'test' ],
                                file(params.modules_testdata_base_path + 'imaging/ome-tiff/cycif-tonsil-cycle1.ome.tif', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("stub - basicpy - ome-tiff") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                                [ id:'test' ],
                                file(params.modules_testdata_base_path + 'imaging/ome-tiff/cycif-tonsil-cycle1.ome.tif', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
