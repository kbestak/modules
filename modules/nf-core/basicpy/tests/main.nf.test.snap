{
    "basicpy - OME-TIFF": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "cycif-tonsil-cycle1.ome-dfp.tiff:md5,ed8e02bbd04d694ff5a5f289f48f97ab"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "cycif-tonsil-cycle1.ome-ffp.tiff:md5,e1a1a2398ef5b76110db6114541f7511"
                    ]
                ],
                "2": [
                    "versions.yml:md5,1391fd5a5745a20016f9824467273113"
                ],
                "dfp": [
                    [
                        {
                            "id": "test"
                        },
                        "cycif-tonsil-cycle1.ome-dfp.tiff:md5,ed8e02bbd04d694ff5a5f289f48f97ab"
                    ]
                ],
                "ffp": [
                    [
                        {
                            "id": "test"
                        },
                        "cycif-tonsil-cycle1.ome-ffp.tiff:md5,e1a1a2398ef5b76110db6114541f7511"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1391fd5a5745a20016f9824467273113"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-11T14:56:34.908774735"
    },
    "basicpy - OME-TIFF - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.-dfp.tiff:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.-ffp.tiff:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,1391fd5a5745a20016f9824467273113"
                ],
                "dfp": [
                    [
                        {
                            "id": "test"
                        },
                        "test.-dfp.tiff:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "ffp": [
                    [
                        {
                            "id": "test"
                        },
                        "test.-ffp.tiff:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1391fd5a5745a20016f9824467273113"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-11T14:56:44.208326285"
    }
}